src/components/table/template.test.js

```diff
@@ -0,0 +1,362 @@
+          {
+            'text': '£85',
+            'format': 'numeric',
+            'colspan': 2
+          }
+        ],
+        [
+          {
+            'text': 'February'
+          },
+          {
+            'text': '£165',
+            'format': 'numeric'
+          }
+        ]
+      ]
+    })
+    const $component = $('.govuk-c-table')
+    const $tableCell = $component.find('.govuk-c-table__body .govuk-c-table__row .govuk-c-table__cell')
+
+    expect($tableCell.eq(1).attr('colspan')).toEqual('2')
+  })
+
+  it('renders cell `rowspan` attribute correctly', () => {
+    const { $ } = render('table', {
+      'rows': [
+        [
+          {
+            'text': 'January'
+          },
+          {
+            'text': '£85',
+            'format': 'numeric',
+            'rowspan': 2
+          }
+        ],
+        [
+          {
+            'text': 'February'
+          },
+          {
+            'text': '£165',
+            'format': 'numeric'
+          }
+        ]
+      ]
+    })
+    const $component = $('.govuk-c-table')
+    const $tableCell = $component.find('.govuk-c-table__body .govuk-c-table__row .govuk-c-table__cell')
+
+    expect($tableCell.eq(1).attr('rowspan')).toEqual('2')
+  })
+
+  it('renders cell `format` attribute correctly', () => {
+    const { $ } = render('table', {
+      'rows': [
+        [
+          {
+            'text': 'January'
+          },
+          {
+            'text': '£85',
+            'format': 'numeric'
+          }
+        ],
+        [
+          {
+            'text': 'February'
+          },
+          {
+            'text': '£165',
+            'format': 'numeric'
+          }
+        ]
+      ]
+    })
+    const $component = $('.govuk-c-table')
+    const $tableCell = $component.find('.govuk-c-table__body .govuk-c-table__cell')
+
+    expect($tableCell.eq(1).attr('class')).toMatch('govuk-c-table__cell--numeric')
+  })
+
+  it('renders first cell in every row as a <th> element with correct `govuk-c-table__header` class and `scope=row` attribute', () => {
+    const { $ } = render('table', {
+      'firstCellIsHeader': true,
+      'rows': [
+        [
+          {
+            'text': 'January'
+          },
+          {
+            'text': '£85',
+            'format': 'numeric'
+          },
+          {
+            'text': '£95',
+            'format': 'numeric'
+          }
+        ],
+        [
+          {
+            'text': 'February'
+          },
+          {
+            'text': '£75',
+            'format': 'numeric'
+          },
+          {
+            'text': '£55',
+            'format': 'numeric'
+          }
+        ]
+      ]
+    })
+    const $component = $('.govuk-c-table')
+    const $tableRow = $component.find('.govuk-c-table .govuk-c-table__body .govuk-c-table__row')
+
+    expect($tableRow.eq(0).children().eq(0).get(0).tagName).toEqual('th')
+    expect($tableRow.eq(0).children().eq(0).hasClass('govuk-c-table__header')).toBeTruthy()
```

Comment:
We could swap this around and make this a bit easier to read by doing

Instead of 


```js
expect($tableRow.eq(0).children().eq(0).get(0).tagName).toEqual('th')
 expect($tableRow.eq(0).children().eq(0).hasClass('govuk-c-table__header')).toBeTruthy()
```

try

```js
const $tableBody = $component.find('.govuk-c-table .govuk-c-table__body')

const $firstTableRow = $tableBody.find('.govuk-c-table__row:first-child')
const $firstTableHeader = $firstTableRow.find('govuk-c-table__header:first-child')
expect($firstTableHeader.get(0).tagName).toEqual('th')

const $lastTableRow = $tableBody.find('.govuk-c-table__row:last-child')
const $lastTableHeader = $lastTableRow.find('govuk-c-table__header:last-child')
expect($lastTableHeader.get(0).tagName).toEqual('th')

```